app/js/PlaceholderController.js

```diff
@@ -0,0 +1,296 @@
+            "properties": {
+                "userId": { "type": "number" },
+                "id":     { "type": "string" },
+                "title":  { "type": "string" },
+                "body":   { "type": "string" }
+            }
+        }
+    };
+
+    $scope.postSchema = $scope.postsSchema.items;
+
+    $scope.postsUiSchema = {
+        "type": "Control",
+        "scope": { "$ref": "#" },
+        "columns": [
+            {
+                "label": "Title",
+                "scope": {
+                    "$ref": "#/items/properties/title"
+                }
+            },
+            {
+                "label": "User",
+                "scope": {
+                    "$ref": "#/items/properties/userId"
+                },
+                "href": {
+                    "url": "/placeholder-users"
+                }
+            }
+        ]
+    };
+
+    //
+    // Users --
+    //
+
+    $scope.userUiSchema = {
+        "type": "VerticalLayout",
+        "elements": [{
+            "type": "Control",
+            "scope": { "$ref": "#/properties/name" }
+        }]
+    };
+
+    $scope.usersUiSchema = {
+        "type": "Control",
+        "scope": { "$ref": "#" },
+        "columns": [
+            {
+                "label": "Name",
+                "scope": {
+                    "$ref": "#/items/properties/name"
+                }
+            },
+            {
+                "label": "Username",
+                "scope": {
+                    "$ref": "#/items/properties/username"
+                }
+            },
+            {
+                "label": "Email",
+                "scope": {
+                    "$ref": "#/items/properties/email"
+                },
+                "href": {
+                    url: "/placeholder-users",
+                    scope: {
+                        "$ref": '/items/properties/id'
+                    }
+                }
+            }
+        ]
+    };
+
+    $scope.usersSchema = {
+        "type": "array",
+        "items": {
+            "type": "object",
+            "properties": {
+                "id":       {"type": "number"},
+                "name":     {"type": "string"},
+                "username": {"type": "string"},
+                "email":    {"type": "string"}
+            }
+        }
+    };
+
+    $scope.userSchema = $scope.usersSchema.items;
+
+    //
+    // Comments --
+    //
+    $scope.commentUiSchema = {
+        "type": "VerticalLayout",
+        "elements": [
+            {
+                "type": "Control",
+                "scope": { "$ref": "#/properties/email" }
+            },
+            {
+                "type": "Control",
+                "scope": { "$ref": "#/properties/body" }
+            },
+            {
+                "type": "ReferenceControl",
+                "scope": { "$ref": "#/properties/postId" },
+                "href": {
+                    "url": "/placeholder-users",
+                    "title": "Go to post"
```

Comment:
This should be called ```label``` instead of ```title``` to be consistent with other controls. Also it seems that this property is not read anywhere at the moment.